litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go

```diff
@@ -169,10 +170,13 @@ func ClusterOperations(manifest string, requestType string, namespace string) (*
 		return nil, err
 	}
 
-	// Obtain REST interface for the GVR
-	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+	//check for agent scope to ensure namespace for dynamic client is in compliance with the enforced RBAC.
+	if AgentScope == "namespace" {
+		// Obtain REST interface for the GVR
+		dr = dynamicClient.Resource(mapping.Resource).Namespace(AgentNamespace)
+	} else if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
 		// namespaced resources should specify the namespace
-		dr = dynamicClient.Resource(mapping.Resource).Namespace(namespace)
+		dr = dynamicClient.Resource(mapping.Resource).Namespace(obj.GetNamespace())
```

Comment:
A problem will arise for "obj.GetNamespace()" here when manifest doesn't have namespace field. It will not be default also because it is reading from YAML.

Reply:
That's why we need namespace to be passed from `clusterConnection` subscription if no namespace field is there in the manifest

Reply:
Yeah that error i am getting so will be making some changes for it..